tests/build/jvm-build.go

```diff
@@ -35,22 +36,47 @@ var (
 var _ = framework.JVMBuildSuiteDescribe("JVM Build Service E2E tests", Label("jvm-build"), func() {
 	defer GinkgoRecover()
 
-	var testNamespace, prGeneratedName, applicationName, componentName, outputContainerImage string
+	var testNamespace, applicationName, componentName, outputContainerImage string
+	var componentPipelineRun v1beta1.PipelineRun
 	var timeout, interval time.Duration
 
 	f, err := framework.NewFramework()
 	Expect(err).NotTo(HaveOccurred())
 
 	AfterAll(func() {
+		abList, err := f.JvmbuildserviceController.ListArtifactBuilds(testNamespace)
+		if err != nil {
+			klog.Infof("got error fetching artifactbuilds: %s", err.Error())
+		}
+
+		dbList, err := f.JvmbuildserviceController.ListDependencyBuilds(testNamespace)
+		if err != nil {
+			klog.Infof("got error fetching dependencybuilds: %s", err.Error())
+		}
+
 		if CurrentSpecReport().Failed() {
+			var testLogsDir string
+			artifactDir := os.Getenv("ARTIFACT_DIR")
+			var storeLogsInFiles bool
+
+			if artifactDir != "" {
+				testLogsDir = fmt.Sprintf("%s/jvm-build-service-test", artifactDir)
```

Comment:
From now on all jvm-build-service-test related logs should be stored in CI in `<JOB_ID>/artifacts/redhat-appstudio-e2e/redhat-appstudio-e2e/artifacts/`

Reply:
@gabemontero kinda bittersweet feeling seeing the test I updated failing :) [here's the link](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_e2e-tests/181/pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e/1562781218133512192/artifacts/redhat-appstudio-e2e/redhat-appstudio-e2e/artifacts/jvm-build-service-test/)

Reply:
I see a similar flake I saw in our post stand up discussion that I saw in Michal's build-service PR (potential upstream maven access):

```
2022-08-25 13:54:47,101 WARN  [com.red.hac.sid.res.MavenProxyResource] (executor-thread-0) Failed retrieving artifact org/sonarsource/scanner/gradle/sonarqube-gradle-plugin/3.3/sonarqube-gradle-plugin-3.3.module, waiting 4000 seconds
2022-08-25 13:54:51,107 ERROR [com.red.hac.sid.res.MavenProxyResource] (executor-thread-0) Failed to load sonarqube-gradle-plugin-3.3.module: javax.ws.rs.NotFoundException: HTTP 404 Not Found
	at com.redhat.hacbs.sidecar.resources.MavenProxyResource.getGavInputStream(MavenProxyResource.java:151)
	at com.redhat.hacbs.sidecar.resources.MavenProxyResource.get(MavenProxyResource.java:106)
	at com.redhat.hacbs.sidecar.resources.MavenProxyResource$quarkusrestinvoker$get_530612623f3f48cd230d326175ee0e0f03e9b6d2.invoke(Unknown Source)
	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:7)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:141)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

this one is on the DB side (client side) vs. the artifact cache side

Reply:
and hence we have a failed dependencybuild:  https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_e2e-tests/181/pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e/1562781218133512192/artifacts/redhat-appstudio-e2e/redhat-appstudio-e2e/artifacts/jvm-build-service-test/appstudio-e2e-test-db-ea268de30de6bab363b6ccea90e4628c.yaml confirmed with the new debug you are adding here

Reply:
another dependencybuild was still building when the test gave up:  https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_e2e-tests/181/pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e/1562781218133512192/artifacts/redhat-appstudio-e2e/redhat-appstudio-e2e/artifacts/jvm-build-service-test/appstudio-e2e-test-db-8b31ad91766ce8c95fda3b3fb485203c.yaml

Reply:
another db failure: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_e2e-tests/181/pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e/1562781218133512192/artifacts/redhat-appstudio-e2e/redhat-appstudio-e2e/artifacts/jvm-build-service-test/appstudio-e2e-test-db-55df1004d27dabdc12e4db533fd222ed.yaml

Reply:
I do wonder if that 404 on the gradle dependency means that the builder image config needs to be updated ... a la https://github.com/redhat-appstudio/infra-deployments/blob/main/components/build/jvm-build-service/system-config.yaml#L9-L15 but I have little expertise there

More of a @stuartdouglas or @dwalluck question when they get back from PTO next week.